### ISSUE TITLE
Get all php version fails on MacOS Catalina 10.15.5

### ISSUE THREAD

**Issue by simontegg** [totara/totara-docker-dev#194]
The set_hosts.sh does not write any hosts

**To Reproduce**
`./tools/set_hosts.sh

**Expected behavior**
Write the hosts

**Additional context**
The culprit is L38 `php_versions=($(cat "$project_path/compose/php.yml" | sed -r -n 's/.*\php-([0-9]).([0-9])[^:]*:/\1\2/p' | uniq | sort))' 

sed -r is an illegal operation. Replacing with sed -E fixes the problem

**Comment by derschatta** [comment]
also the date call

`date=$(date --iso-8601)`

fails with

```
date: illegal option -- -
usage: date [-jnRu] [-r seconds|file] [-v[+|-]val[ymwdHMS]]
            [-I[date | hours | minutes | seconds]]
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
```

on MacOS

**Comment by derschatta** [comment]
fixed in #222
